demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp

```diff
@@ -0,0 +1,196 @@
+                                                 static_cast<int>((x_max - x_min) * scaling_x),
+                                                 static_cast<int>((y_max - y_min) * scaling_y)),
+                                        conf);
+            }
+    }
+};
+
+GAPI_OCV_KERNEL_ST(OCVPersonTrack, custom::PersonTrack, Tracker) {
+    static void setup(const cv::GMatDesc&,
+                      const cv::GArrayDesc&,
+                            std::shared_ptr<Tracker>& tracker_action,
+                      const cv::GCompileArgs& compileArgs) {
+        TrackerParams tracker_action_params;
+        tracker_action_params.min_track_duration = 8;
+        tracker_action_params.forget_delay = 150;
+        tracker_action_params.affinity_thr = 0.9f;
+        tracker_action_params.averaging_window_size_for_rects = 5;
+        tracker_action_params.averaging_window_size_for_labels = 1;
+        tracker_action_params.bbox_heights_range = cv::Vec2f(10, 2160);
+        tracker_action_params.drop_forgotten_tracks = false;
+        tracker_action_params.max_num_objects_in_track = std::numeric_limits<int>::max();
+        tracker_action = std::make_shared<Tracker>(tracker_action_params);
+    }
+    static void run(const cv::Mat& frame,
+                    const TrackedObjects& actions,
+                          TrackedObjects& tracked_actions,
+                          Tracker& tracker_action) {
+        tracker_action.Process(frame, actions);
+        tracked_actions = tracker_action.TrackedDetectionsWithLabels();
+    }
+};
+
+GAPI_OCV_KERNEL_ST(OCVPreprocessing, custom::ActionRecognisePreprocessing, ARStateParams) {
+    static void setup(const cv::GMatDesc&,
+                      const cv::GArrayDesc&,
+                      const cv::GOpaqueDesc&,
+                      const cv::Scalar&,
+                            std::shared_ptr<ARStateParams>& ar_state) {
+        std::list<cv::Mat> frames = {};
+        ar_state = std::make_shared<ARStateParams>(ARStateParams{frames, 0});
+    }
+
+    static void run(const cv::Mat& in_frame,
+                    const TrackedObjects& tracked_persons,
+                    const std::shared_ptr<int>& runtime_id,
+                    const cv::Scalar& net_size,
+                          std::vector<cv::Mat> &prepared_mat,
+                    ARStateParams& ar_state) {
+        const int channels = int(net_size[0]);
+        const int batch = int(net_size[1]);
+        const int height = int(net_size[2]);
+        const int width = int(net_size[3]);
+        cv::Size sz_dst(height, width);
+        if (ar_state.stored_frames.size() < size_t(batch)) {
+            ar_state.stored_frames.emplace_back(in_frame.clone());
+        } else {
+            if(prepared_mat.size() == 0) {
+                cv::Mat mat_to_process;
+                mat_to_process.create({1, channels, batch, height, width}, CV_32F);
+                // FIXME: Why is GArray result?
```

Comment:
Probably it was done to avoid the automatic preprocessing which kicks in here.

Reply:
Yes. It is.